ngraph/test/op.cpp

```diff
@@ -114,21 +114,3 @@ TEST(op, variant)
     Ship& node_ship = as_type_ptr<VariantWrapper<Ship>>(node_var_ship)->get();
     EXPECT_EQ(&node_ship, &ship);
 }
-
-// TODO: Need to mock Node, Op etc to be able to unit test functions like replace_node().
-// Mocking them directly isn't possible because google test requires methods to be
-// non-virtual. For non-virtual methods we will need to templatize these classes and call using
-// different template argument between testing and production.
-/*
-TEST(op, provenance_replace_node)
-{
-    class MockOp: public op::Op
-    {
-        MOCK_CONST_METHOD1(copy_with_new_args, std::shared_ptr<Node>(const NodeVector& new_args));
-        MOCK_CONST_METHOD1(get_users, NodeVector (bool check_is_used)); // This can't be mocked as
-                                                                        // it's non-virtual
-    };
-
-    ::testing::NiceMock<MockOp> mock_op;
-}
-*/
```

Comment:
Why you removed this test?

Reply:
This test were commented. I just removed redundant commented code.

Reply:
Is there ticket that will remind to enable such test in future?

Reply:
No this is nGraph mechanism which is not used in the OpenVINO.

Reply:
But if there is still such mechanism should it be covered by tests even if now it is unused in OpenVINO? If you consider mechanism useless should it be removed with rest as well (or you're doing it right now removing `copy_with_new_args`)?

Reply:
At the current moment I removed all places with `copy_with_new_args` these commented code was a PoC to mock node object and this method. The new method `clone_with_new_inputs` is covered by tests.